src/OutputWriters/bit_rounding.jl

```diff
@@ -0,0 +1,52 @@
+using BitInformation
+
+struct BitRounding{K}
+    keepbits :: K
+end
+
+# number of keepbits (mantissa bits) for each variable
+default_bit_rounding(::Val{name}) where name = 23   # single precision default
+default_bit_rounding(::Val{:u}) = 2
+default_bit_rounding(::Val{:v}) = 2
+default_bit_rounding(::Val{:w}) = 2
+default_bit_rounding(::Val{:T}) = 7
+default_bit_rounding(::Val{:S}) = 16                # 12 at the surface, 16 deep ocean
```

Comment:
This is interesting. Why is there a difference between T, S? Is this specific to the simulation that this was tested on, or can we be sure this is valid for all simulations, past climates, future climates, idealized simulations at other resolutions, etc?

It seems we need to have default bit rounding for passive tracers.

Reply:
Although relatively robust through time and space, this depends on a lot of things, also whether your unit carries some offset around (e.g. Kelvin vs ˚C, density vs density anomaly). So it's tricky to generalise. I suggest to have some reasonable defaults if someone uses bit rounding (default `nothing` or single precision as you like) but suggest to highlight that this should be checked similar to how I did it here with the bitinformation analysis above.

For global ocean simulations I expect these to be reasonable defaults. I believe for now this is mostly to reduce the filesizes for OMIP simulations

Reply:
True, I'm just not sure that OMIP is going to be the most common use case, so there's a question about what default is appropriate here

Reply:
The OMIP defaults might belong in the `ClimaOcean` setup, perhaps

Reply:
We could set the defaults here as 23 mantissa bits (=Float32 precision, whether you use Float32 or 64) and then lower in ClimaOcean?